pulsar-zookeeper-utils/src/main/java/com/yahoo/pulsar/zookeeper/ZooKeeperDataCache.java

```diff
@@ -143,7 +143,7 @@ public void process(WatchedEvent event) {
         }
     }
 
-    public void close() {
+    public synchronized void close() {
```

Comment:
We are worried about the following potential issue:
Thread 1: Watch fires and calls `ZooKeeperDataCache.process` and enters the `IS_SHUTDOWN_UPDATER.get()` condition.
Thread 2: Calls `close()` and then `clear()`
Thread 1: Finishes `process` after `clear()` is called.
When `close` is synchronized, another thread cannot enter `process` and we can safely `clear` knowing that no more updates will follow.